pennylane/drawer/tape_text.py

```diff
@@ -0,0 +1,241 @@
+    """Adds symbols indicating the extent of a given object."""
+    if len(op.wires) > 1:
+        mapped_wires = [wire_map[w] for w in op.wires]
+        min_w, max_w = min(mapped_wires), max(mapped_wires)
+        layer_str[min_w] = "╭"
+        layer_str[max_w] = "╰"
+
+        for w in range(min_w + 1, max_w):
+            layer_str[w] = "├" if w in mapped_wires else "│"
+
+    return layer_str
+
+
+def _add_op(op, layer_str, wire_map, decimals):
+    """Updates ``layer_str`` with ``op`` operation."""
+    layer_str = _add_grouping_symbols(op, layer_str, wire_map)
+
+    control_wires = op.control_wires
+    for w in control_wires:
+        layer_str[wire_map[w]] += "C"
+
+    label = op.label(decimals=decimals).replace("\n", "")
+    for w in op.wires:
+        if w not in control_wires:
+            layer_str[wire_map[w]] += label
+
+    return layer_str
+
+
+def _add_measurement(m, layer_str, wire_map, decimals):
+    """Updates ``layer_str`` with the ``m`` measurement."""
+    layer_str = _add_grouping_symbols(m, layer_str, wire_map)
+
+    obs_label = "" if m.obs is None else m.obs.label(decimals=decimals).replace("\n", "")
+    meas_label = measurement_label_map[m.return_type](obs_label)
+
+    if len(m.wires) == 0:  # state or probability across all wires
+        for i, s in enumerate(layer_str):
+            layer_str[i] = s + meas_label
+
+    for w in m.wires:
+        layer_str[wire_map[w]] += meas_label
+    return layer_str
+
+
+# pylint: disable=too-many-arguments
+def tape_text(
+    tape, wire_order=None, show_all_wires=False, decimals=None, max_length=100, tape_offset=None
+):
+    """Text based diagram for a Quantum Tape.
+
+    Args:
+        tape (QuantumTape): the operations and measurements to draw
+
+    Keyword Args:
+        wire_order (Sequence[Any]): the order (from top to bottom) to print the wires of the circuit
+        show_all_wires (bool): If True, all wires, including empty wires, are printed.
+        decimals (int): How many decimal points to include when formatting operation parameters.
+            Default ``None`` will omit parameters from operation labels.
+        decimals (Int) : how many decimal points to display in the operation label.  If ``None``,
+            no parameters will be displayed.
+        max_length (Int) : Maximum length of a individual line.  After this length, the diagram will
+            begin anew beneath the previous lines.
+        tape_offset (list[Int]): Used to offset numbering when labelling nested tapes. Used internally for
+            recursive calls.
+
+    Returns:
+        str : String based graphic of the circuit.
+
+    **Example:**
+
+    .. code-block:: python
+
+        with qml.tape.QuantumTape() as tape:
+            qml.QFT(wires=(0,1,2))
+            qml.RX(1.234, wires=0)
+            qml.RY(1.234, wires=1)
+            qml.RZ(1.234, wires=2)
+            qml.Toffoli(wires=(0,1,"aux"))
+
+            qml.expval(qml.PauliZ("aux"))
+            qml.state()
```

Comment:
The change in a4213f4 fixes this, but the current example tape now raises a new error:
```py
with qml.tape.QuantumTape() as tape:
    qml.QFT(wires=(0,1,2))
    qml.RX(1.234, wires=0)
    qml.RY(1.234, wires=1)
    qml.RZ(1.234, wires=2)
    qml.Toffoli(wires=(0,1,"aux"))
    qml.expval(qml.PauliZ("aux"))
    qml.var(qml.PauliZ(0) @ qml.PauliZ(1))
    qml.probs()

qml.execute([tape], qml.device('default.qubit', wires=[0,1,2,'aux']), None)
```
```
ValueError: cannot reshape array of size 0 into shape (0)
```

Reply:
wires are now specified.  I feel like at some point in time we used to be able to call `qml.probs()` and have it broadcast to all wires, but obviously now we can't.

Reply:
> I feel like at some point in time we used to be able to call qml.probs() and have it broadcast to all wires, but obviously now we can't.

I don't think we did, but it is a bit confusing because we do allow `return qml.state()`. In fact, `qml.state()` doesn't allow any wires to be passed, it automatically broadcasts to all wires, to avoid issues with partial tracing.